### PR TITLE
SDI-18 Other personal contacts page

### PR DESCRIPTION
**Pull request by steverendell**
add tests and fixes

### CODE REVIEW

File: integration_tests/mockApis/prisonApi.ts [hunk 1]
```diff
@@ -0,0 +1,76 @@
...
+      urlPath: `/api/persons/${personId}/addresses`,
+    },
+    response: {
+      status: 200,
+      headers: { 'Content-Type': 'application/json;charset=UTF-8' },
+      jsonBody: response,
+    },
+  })
+
+const stubGetPrisonerPhones = ({ personId, response }: { personId: number; response: PrisonApiTelephone[] }) =>
```
Comment (andymarke) [inline integration_tests/mockApis/prisonApi.ts:44]: seems odd we are having to make calls, does this stubbing imply that the main contacts call isn't retrieving address etc? And we would end up making a lot of calls just to display contacts for a single prisoner?

File: server/services/nomisPrisonerService.ts [hunk 2]
```diff
@@ -12,9 +12,69 @@ export type PrisonApiTelephone = components['schemas']['Telephone']
...
 export type PrisonApiEmail = components['schemas']['Email']
 
-export interface Context {
-  username?: string
-  token?: string
+export interface EmailDto {
+  email: string
+}
+
+interface RestrictionDto {
```
Comment (andymarke) [inline server/services/nomisPrisonerService.ts:19]: shouldn't these be generated from the swagger or are these not API responses?

File: server/services/nomisPrisonerService.ts [hunk 3]
```diff
@@ -26,31 +86,28 @@ export default class NomisPrisonerService {
 
   async getPrisonerContacts(context: Context, offenderNo: string): Promise<PrisonApiContacts> {
     const token = await this.hmppsAuthClient.getSystemClientToken(context.username)
-    logger.debug(`getting contact details for ${offenderNo}`)
     return NomisPrisonerService.restClient(token).get<PrisonApiContacts>({
       path: `/api/offenders/${offenderNo}/contacts`,
+      query: 'activeOnly=true',
     })
```
Comment (andymarke) [inline server/services/nomisPrisonerService.ts:92]: I have a feeling the active flag in NOMIS for contacts is nonsense, best to check prod data. I'm sure we saw a prison where every contact was inactive

**Review by andymarke — APPROVED** [review]
LGTM

File: integration_tests/mockApis/prisonApi.ts (hunk 1 above)
Comment (steverendell) [reply integration_tests/mockApis/prisonApi.ts:44]: Thats correct!

File: server/services/nomisPrisonerService.ts (hunk 3 above)
Comment (steverendell) [reply server/services/nomisPrisonerService.ts:92]: oh dear, ok good idea

**Pull request merged**